pkg/modulewriter/modulewriter.go

```diff
@@ -338,3 +366,68 @@ func prepDepDir(depDir string, overwrite bool) error {
 	}
 	return nil
 }
+
+func filterVarsByGraph(vars map[string]interface{}, group config.DeploymentGroup, graph map[string][]config.ModConnection) map[string]interface{} {
+	if graph == nil {
+		return vars
+	}
+
+	groupInputs := make(map[string]bool)
+	for _, mod := range group.Modules {
+		if connections, ok := graph[mod.ID]; ok {
+			for _, conn := range connections {
+				if conn.IsDeploymentKind() {
+					for _, v := range conn.GetSharedVariables() {
+						groupInputs[v] = true
+					}
+				}
+			}
+		}
+	}
+
+	filteredVars := make(map[string]interface{})
+	for key, val := range vars {
+		if _, ok := groupInputs[key]; ok {
+			filteredVars[key] = val
+		}
+	}
+	return filteredVars
+}
+
+func writeDeploymentMetadata(depDir string, metadata deploymentMetadata) error {
+	ghpcDir := filepath.Join(depDir, hiddenGhpcDirName)
+	if _, err := os.Stat(ghpcDir); os.IsNotExist(err) {
+		return fmt.Errorf(
+			"While trying to update the deployment directory at %s, the '.ghpc/' dir could not be found", depDir)
+	}
+
+	metadataFile := filepath.Join(ghpcDir, deploymentMetadataName)
+	f, err := os.OpenFile(metadataFile, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var buf bytes.Buffer
+	encoder := yaml.NewEncoder(&buf)
```

Comment:
nit.
```go
encoder := yaml.NewEncoder(&buf)
defer encoder.Close()
encoder.SetIndent(2)
if err := encoder.Encode(metadata); err != nil {
	return err
}
```

Reply:
Done (along with the more important closing of the file)